.travis.yml

```diff
@@ -11,35 +11,44 @@ services:
 git:
     submodules: false
 
-# Do a shallow submodule fetch
-before_install: git submodule update --init
-
+# Global environment variables
 env:
     global:
         - DIST=xenial
         # Make sure beaver is in the PATH
         - PATH="$(git config -f .gitmodules submodule.beaver.path)/bin:$PATH"
-    matrix:
-        - DMD=dmd1 F=production
-        - DMD=dmd1 F=devel
-        - DMD=dmd-transitional F=production
-        - DMD=dmd-transitional F=devel
 
-install: beaver dlang install
+# Do a shallow submodule fetch
+before_install: git submodule update --init
 
-# Build packages too, this should be done by beaver:
-# https://github.com/sociomantic-tsunami/beaver/issues/34
-script: beaver dlang make && beaver dlang make pkg
+# Create docker images and setup environment to build
+install: beaver dlang install
 
+# Basic config is inherited from the global scope
 jobs:
+    templates:
+        - &test-matrix
+          stage: Test
+          script:
+              - beaver dlang make
+              # Build packages too, this should be done by beaver:
+              # https://github.com/sociomantic-tsunami/beaver/issues/34
+              - beaver dlang make pkg
     include:
+        # Test matrix
+        - <<: *test-matrix
+          env: DMD=dmd1 F=production
```

Comment:
One last request - while we are touching this, maybe you can switch dhtnode to use pinned compiler versions like we do in ocean/swarm/turtle?

Reply:
Sure, but I will do it in another PR.